### ISSUE TITLE
Memory leak in the RequestResponseEndpoint

### ISSUE THREAD

**Issue by ctron** [eclipse/hono#753]
To me it looks like there is a memory leak in the device registry. After running for a while the device registry gets slower and slower. And the protocol adapters fail with 503, originating from the communication with the device registry.

Restarting the pod fixes the issue:

![image](https://user-images.githubusercontent.com/202474/43636358-23b0fffc-9712-11e8-8d63-c37bb2142399.png)

And running leak analysis on the heap dump seems to confirm this:

![image](https://user-images.githubusercontent.com/202474/43636381-3663c21a-9712-11e8-9452-7b51a9a792e0.png)

Further analysis is required how the happens.

**Comment by ctron** [comment]
One possible cause for this could be `org.eclipse.hono.service.amqp.RequestResponseEndpoint.onLinkAttach(ProtonConnection, ProtonSender, ResourceIdentifier)`. It does register with the event bus and unregisters when the link is closed:

~~~java
final MessageConsumer<JsonObject> replyConsumer = vertx.eventBus().consumer(replyToAddress.toString(), message -> {
…
});
…
HonoProtonHelper.setCloseHandler(sender, senderClosed -> {
    logger.debug("client [{}] closed sender link, removing associated event bus consumer [{}]", sender.getName(), replyConsumer.address());
    replyConsumer.unregister();
    if (senderClosed.succeeded()) {
        senderClosed.result().close();
    }
});
sender.open();
~~~

However it doesn't seem to unregister when the session or transport gets closed. Which would happen every 10 minutes, when the service closes the connection due to the token expiration. Or, when the connection breaks due to other reasons.

Also does the method `org.eclipse.hono.service.amqp.AmqpServiceBase.closeExpiredConnection(ProtonConnection)` reset both the close and the disconnect handler.

**Comment by sophokles73** [comment]
@ctron are you going to work on this? Otherwise, I could take a look, if you are stuck in the middle of something else. If you want me to try to fix this, simply assign the issue to me ...

**Comment by ctron** [comment]
Fixed with PR #760
